<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from my_robot.xacro                 | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="myFirst" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.2"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.05"/>
    </visual>
  </link>
  <link name="link_1">
    <visual>
      <geometry>
        <box size="0.2 0.1 0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.05"/>
      <material name="blue">
        <color rgba="0 0 .8 1"/>
      </material>
    </visual>
  </link>
  <link name="link_2">
    <visual>
      <geometry>
        <cylinder length="0.3" radius="0.025"/>
      </geometry>
      <origin rpy="1.57 0 0" xyz="0 -0.15 0"/>
      <material name="yellow">
        <color rgba="1 1 0 1"/>
      </material>
    </visual>
  </link>
  <link name="link_3">
    <visual>
      <geometry>
        <cylinder length="0.05" radius="0.025"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <link name="link_4">
    <visual>
      <geometry>
        <cylinder length="0.3" radius="0.025"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.15"/>
      <material name="yellow"/>
    </visual>
  </link>
  <link name="link_5">
    <visual>
      <geometry>
        <cylinder length="0.05" radius="0.025"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="white"/>
    </visual>
  </link>
  <link name="link_6">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.025"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.05"/>
      <material name="yellow"/>
    </visual>
  </link>
  <joint name="joint_0" type="fixed">
    <parent link="base_link"/>
    <child link="link_1"/>
    <origin rpy="0 0 0" xyz="0 0 0.1"/>
  </joint>
  <joint name="joint_1" type="revolute">
    <parent link="base_link"/>
    <child link="link_1"/>
    <limit effort="100" lower="-3.14" upper="3.14" velocity="180"/>
    <origin rpy="0 0 0.00174532922222" xyz="0 0 0.1"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="joint_2" type="revolute">
    <parent link="link_1"/>
    <child link="link_2"/>
    <limit effort="100" lower="-3.14" upper="3.14" velocity="180"/>
    <origin rpy="-1.5707963 0.00349065844444 0" xyz="0.1001 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="joint_3" type="revolute">
    <parent link="link_2"/>
    <child link="link_3"/>
    <limit effort="100" lower="-3.14" upper="3.14" velocity="180"/>
    <origin rpy="3.1415926 0 -1.57603228767" xyz="0 -0.3002 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="joint_4" type="revolute">
    <parent link="link_3"/>
    <child link="link_4"/>
    <limit effort="100" lower="-3.14" upper="3.14" velocity="180"/>
    <origin rpy="-1.5707963 0.00698131688889 0" xyz="0.0203 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="joint_5" type="revolute">
    <parent link="link_4"/>
    <child link="link_5"/>
    <limit effort="100" lower="-3.14" upper="3.14" velocity="180"/>
    <origin rpy="1.5707963 -0.00872664611111 0" xyz="0 0 -0.2996"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="joint_6" type="revolute">
    <parent link="link_5"/>
    <child link="link_6"/>
    <limit effort="100" lower="-3.14" upper="3.14" velocity="180"/>
    <origin rpy="1.5707963 -0.0104719753333 0" xyz="0 -0.0995 0"/>
    <axis xyz="0 0 1"/>
  </joint>
</robot>

